gtsam/discrete/DecisionTree-inl.h

```diff
@@ -309,7 +312,7 @@ namespace gtsam {
       }
 #endif
       size_t index = x.at(label_);
-      NodePtr child = branches_[index];
+      NodePtr child = branches_.at(index);
```

Comment:
Why did this need to be changed?

Reply:
Was getting segfaults over here. This change means we now get a proper error message about a vector indexing problem.